File: Medium/Minimum Cost Selection/minimum-cost-selection.py
class Solution:

    def minCost(self, mat):
        """code here"""
        n = len(mat)
        choice1 = mat[0][0]
        choice2 = mat[0][1]
        choice3 = mat[0][2]
        
        for i in range(1 , n):
            cur1 = mat[i][0] + min(choice2 , choice3)
            cur2 = mat[i][1] + min(choice1 , choice3)
            cur3 = mat[i][2] + min(choice1 , choice2)
            
            choice1 = cur1
            choice2 = cur2
            choice3 = cur3
        
        return min(choice1 , choice2 , choice3)

File: Medium/Minimum Cost Selection/minimum-cost-selection.cpp
class Solution {
  public:
    int minCost(vector<vector<int>>& mat) {
        // code here
        int n = mat.size();
        int c0 = mat[0][0];
        int c1 = mat[0][1];
        int c2 = mat[0][2];
        
        for (int i = 1 ; i < n ; i++){
            int t0 = mat[i][0] + min(c1 , c2);
            int t1 = mat[i][1] + min(c0 , c2);
            int t2 = mat[i][2] + min(c1 , c0);
            
            c0 = t0;
            c1 = t1;
            c2 = t2;
        }
        
        return min({c0 , c1 , c2});
    }
};